src/gateway.h

```diff
@@ -54,15 +57,20 @@ void gateway__leader_close(struct gateway *g, int reason);
 
 /**
  * Asynchronous request to handle a client command.
+ *
+ * We also use the handle as a place to save request-scoped data that we need
+ * to access from a callback.
  */
 typedef void (*handle_cb)(struct handle *req, int status, int type);
 struct handle
 {
-	void *data; /* User data */
-	int type;   /* Request type */
+	void *data;                   /* User data */
+	int type;                     /* Request type */
 	struct gateway *gateway;
 	struct buffer *buffer;
 	struct cursor cursor;
+	uint64_t db_id;               /* For use by prepare callback */
```

Comment:
Nit: I'd use `size_t` as type here, which is what is used internally for IDs of registered objects. The fixed size types are used when serializing those values for sending them over the wire.